### PR TITLE
wrangler: add AI related commands

### PR DESCRIPTION
**Pull request by edevil**
**What this PR solves / how to test:**

This PR adds the CLI commands related to AI functionality.

**Author has included the following, where applicable:**

- [x] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

File: packages/wrangler/src/__tests__/ai.test.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+	const std = mockConsoleMethods();
+
+	beforeEach(() => {
+		// @ts-expect-error we're using a very simple setTimeout mock here
+		jest.spyOn(global, "setTimeout").mockImplementation((fn, _period) => {
+			setImmediate(fn);
+		});
+		setIsTTY(true);
+	});
```
Comment (penalosa) [inline packages/wrangler/src/__tests__/ai.test.ts:79]: Why is this needed?

File: packages/wrangler/src/index.ts [hunk 2]
```diff
@@ -451,6 +452,11 @@ export function createCLIParser(argv: string[]) {
 	});
 
 	// ai
+	wrangler.command("ai", "🤖 Interact with AI models", (aiYargs) => {
+		return ai(aiYargs.command(subHelp));
+	});
+
+	// constellation
```
Comment (penalosa) [inline packages/wrangler/src/index.ts:459]: Is the intention to leave in constellation commands?

File: packages/wrangler/src/ai/listCatalog.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		const entries = await listCatalogEntries(accountId);
+
+		if (json) {
+			logger.log(JSON.stringify(entries, null, 2));
+		} else {
+			logger.table(
+				entries.map((entry) => ({
+					model: entry.id,
+					name: entry.name,
+					description: truncate(entry.description ?? "", 100),
```
Comment (penalosa) [inline packages/wrangler/src/ai/listCatalog.tsx:28]: Could this use the user's terminal size to decide on the appropriate truncation?

**Review by penalosa — COMMENTED** [review]
Looking good!
- It would be good to see a test of the truncation behaviour
- It would be good to see some tests of the recursive fetching behaviour

File: packages/wrangler/src/index.ts (hunk 2 above)
Comment (edevil) [reply packages/wrangler/src/index.ts:459]: Yes, they will co-exist for now.

File: packages/wrangler/src/__tests__/ai.test.ts (hunk 1 above)
Comment (edevil) [reply packages/wrangler/src/__tests__/ai.test.ts:79]: It was present on other tests on which I based this, I assumed setTimeout() was being used somewhere in the code that the tests called. I'll remove it.

Comment (edevil) [reply packages/wrangler/src/__tests__/ai.test.ts]: Removed.

File: packages/wrangler/src/ai/listCatalog.tsx (hunk 3 above)
Comment (edevil) [reply packages/wrangler/src/ai/listCatalog.tsx:28]: `process.stdout.columns` seems undefined in tests. It also seems a bit unnecessary IMHO. :)

**Comment by RamIdeas** [comment]
Rerunning the windows test job which failed

**Review by penalosa — APPROVED** [review]
